homeassistant/components/sensor/discogs.py

```diff
@@ -40,32 +43,75 @@
     """Set up the Discogs sensor."""
     import discogs_client
 
-    name = config.get(CONF_NAME)
-    token = config.get(CONF_TOKEN)
+    token = config[CONF_TOKEN]
 
     try:
-        discogs = discogs_client.Client(SERVER_SOFTWARE, user_token=token)
-        identity = discogs.identity()
+        discogs_client = discogs_client.Client(
+            SERVER_SOFTWARE, user_token=token)
     except discogs_client.exceptions.HTTPError:
         _LOGGER.error("API token is not valid")
         return
 
-    async_add_entities([DiscogsSensor(identity, name)], True)
+    async_add_entities([
```

Comment:
I'm a bit in the dark about the exact difference between using add_entities and async_add_entities.
Used async here before because they don't rely on each other or any order to be instantiated.